src/assets/wise5/authoringTool/components/component-type-selector/component-type-selector.component.spec.ts

```diff
@@ -16,14 +17,37 @@ let userService: UserService;
 describe('ComponentTypeSelectorComponent', () => {
   beforeEach(async () => {
     TestBed.configureTestingModule({
-      imports: [
-        ComponentTypeSelectorComponent,
-        ComponentTypeServiceModule,
-        StudentTeacherCommonServicesModule
-      ],
-      providers: [provideHttpClient()]
+      imports: [ComponentTypeSelectorComponent, StudentTeacherCommonServicesModule],
+      providers: [
+        MockProviders(ComponentTypeService, ConfigService, UserService),
+        provideHttpClient()
+      ]
     });
     fixture = TestBed.createComponent(ComponentTypeSelectorComponent);
+    const componentTypeService = TestBed.inject(ComponentTypeService);
+    spyOn(componentTypeService, 'getComponentTypes').and.returnValue([
+      { type: 'AiChat', name: 'AI Chat' },
+      { type: 'Animation', name: 'Animation' },
+      { type: 'AudioOscillator', name: 'Audio Oscillator' },
+      { type: 'ConceptMap', name: 'Concept Map' },
+      { type: 'DialogGuidance', name: 'Dialog Guidance' },
+      { type: 'Discussion', name: 'Discussion' },
+      { type: 'Draw', name: 'Draw' },
+      { type: 'Embedded', name: 'Embedded' },
+      { type: 'Graph', name: 'Graph' },
+      { type: 'Label', name: 'Label' },
+      { type: 'Match', name: 'Match' },
+      { type: 'MultipleChoice', name: 'Multiple Choice' },
+      { type: 'OpenResponse', name: 'Open Response' },
+      { type: 'OutsideURL', name: 'Outside URL' },
+      { type: 'PeerChat', name: 'Peer Chat' },
+      { type: 'HTML', name: 'HTML' },
+      { type: 'ShowGroupWork', name: 'Show Group Work' },
+      { type: 'ShowMyWork', name: 'Show My Work' },
+      { type: 'Summary', name: 'Summary' },
+      { type: 'Table', name: 'Table' }
+    ]);
```

Comment:
Perhaps this could be done without hard-coding. Because we'll need to update whenever a component type is added, removed, or edited, correct?